analyzer/consensus/consensus.go

```diff
@@ -376,6 +378,22 @@ func (m *processor) queueEpochInserts(batch *storage.QueryBatch, data *consensus
 	return nil
 }
 
+// Adapted from https://github.com/oasisprotocol/oasis-core/blob/master/go/consensus/api/transaction/transaction.go#L58
+func unpackTxBody(t *transaction.Transaction) (interface{}, error) {
```

Comment:
method registry is how oasis-core does it, so I think it's fine to continue doing it that way in nexus. we can move them to a different package though, as these are good reasons to keep them separate from the mostly auto-generated vendored type packages.

Reply:
Good points; updated as you suggested! 

Also, I went through the types again and found that several cobalt tx types are incompatible with later tx types used from Damask onwards. I added a separate mapping for those.

Reply:
aw

Reply:
Oop that was (un)lucky timing 😅 

I'd be open to using the registry method as well, it would just be switching the `map[string]interface{}` for a `sync.Map` and reintroducing the `NewMethodName` calls here in this file, right? Though we'd need two maps, one for cobalt and one for damask+eden